src/components/Collections/CollectionShare/publicShare/beforeShare.tsx

```diff
@@ -1,26 +1,32 @@
-import { Box, Typography } from '@mui/material';
-import { FlexWrapper } from 'components/Container';
+import { Stack, Typography } from '@mui/material';
 import { GalleryContext } from 'pages/gallery';
 import { AppContext } from 'pages/_app';
 import React, { useContext, useState } from 'react';
 import { t } from 'i18next';
 import {
     createShareableURL,
     deleteShareableURL,
+    updateShareableURL,
 } from 'services/collectionService';
-import { Collection, PublicURL } from 'types/collection';
+import { Collection, PublicURL, UpdatePublicURL } from 'types/collection';
 import { handleSharingErrors } from 'utils/error/ui';
-import PublicShareSwitch from './switch';
+import { EnteMenuItem } from 'components/Menu/menuItem';
+import PublicIcon from '@mui/icons-material/Public';
 interface Iprops {
+    publicShareProp;
     collection: Collection;
     publicShareActive: boolean;
     setPublicShareProp: (value: PublicURL) => void;
+    setIsFirstShareProp: (value: boolean) => void;
 }
+import LinkIcon from '@mui/icons-material/Link';
+import { EnteMenuItemGroup } from 'components/Menu/menuItemGroup';
 
-export default function PublicShareControl({
+export default function BeforeShare({
```

Comment:
Can Call it `EnablePublicShareOptions`?

also, in general, lets try to use `publicShare` for things refering to Public Sharing to avoid confusion with `EmailShare`

check other places too where, you may have used just `share` and update them too for clarity